### PR TITLE
DI-5304: create api client integration for cart

### PR DESCRIPTION
**Pull request by I2olanD**
- [x] I Accept [Individual Contributor License Agreement](https://github.com/DivanteLtd/next/blob/master/CLA.md)

### CODE REVIEW

**Comment by filrak** [comment]
Please accept CLA :)

**Comment by I2olanD** [comment]
> Please accept CLA :)

done :)

**Review by lukeromanowicz — CHANGES_REQUESTED** [review]
The code looks good to me but we need to improve test coverage

File: packages/about-you/api-client/tests/api/removeFromCart/index.spec.ts [hunk 1]
```diff
@@ -0,0 +1,18 @@
+import removeFromCart from '../../../src/api/removeFromCart';
+import { apiClient } from '../../../src/index';
+
+const mockDeleteItem = jest.fn((params) => params);
+
+apiClient.basket.deleteItem = mockDeleteItem;
```
Comment (lukeromanowicz) [inline packages/about-you/api-client/tests/api/removeFromCart/index.spec.ts]: This isn't really how you should mock apiClient. I suggest to use jest.mock instead.

File: packages/about-you/api-client/tests/api/clearCart/index.spec.ts [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+            key: 'test'
+          }
+        ]
+      }
+    }));
+
+    const mockClearCart = jest.fn((params) => params);
+
+    // @ts-ignore
+    apiClient.basket.get = mockGetBasket;
```
Comment (lukeromanowicz) [inline packages/about-you/api-client/tests/api/clearCart/index.spec.ts]: This isn't really how you should mock apiClient. I suggest to use jest.mock instead.

**Review by lukeromanowicz — CHANGES_REQUESTED** [review]

**Review by lukeromanowicz — APPROVED** [review]
I am truly impressed by how quickly you have adapted to this way of writing unit tests. Awesome job @I2olanD!

**Review by filrak — APPROVED** [review]

**Review by andrzejewsky — APPROVED** [review]

**Pull request merged**